include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+        if not isinstance(vertex_or_edge, (_mgp_mock.Vertex, _mgp_mock.Edge)):
+            raise TypeError(f"Expected _mgp_mock.Vertex or _mgp_mock.Edge, got {type(vertex_or_edge)}")
+
+        self._len = None
+        self._vertex_or_edge = vertex_or_edge
+
+    def __deepcopy__(self, memo):
+        # This is the same as the shallow copy, as the underlying C API should
+        # not support deepcopy. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Edge and _mgp_mock.Vertex types as they are actually references
+        # to graph elements and not proper values.
+        return Properties(self._vertex_or_edge)
+
+    def get(self, property_name: str, default=None) -> object:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        try:
+            return self[property_name]
+        except KeyError:
+            return default
+
+    def set(self, property_name: str, value: object) -> None:
+        if nx.is_frozen(self._vertex_or_edge.graph):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        self[property_name] = value
+
+    def items(self) -> typing.Iterable[Property]:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        vertex_or_edge_props = (
+            self._vertex_or_edge.graph.nodes[self._vertex_or_edge.id]
+            if isinstance(self._vertex_or_edge, _mgp_mock.Vertex)
+            else self._vertex_or_edge.graph.edges[self._vertex_or_edge.edge]
+        )
+
+        for key, value in vertex_or_edge_props.items():
+            if key in ("label", "type"):
+                continue
+
+            yield Property(key, value)
+
+    def keys(self) -> typing.Iterable[str]:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        for item in self.items():
+            yield item.name
+
+    def values(self) -> typing.Iterable[object]:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        for item in self.items():
+            yield item.value
+
+    def __len__(self) -> int:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        if self._len is None:
+            self._len = sum(1 for _ in self.items())
+
+        return self._len
+
+    def __iter__(self) -> typing.Iterable[str]:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        for item in self.items():
+            yield item.name
+
+    def __getitem__(self, property_name: str) -> object:
+        if not self._vertex_or_edge.is_valid():
+            raise InvalidContextError()
+
+        vertex_or_edge_props = (
+            self._vertex_or_edge.graph.nodes[self._vertex_or_edge.id]
+            if isinstance(self._vertex_or_edge, _mgp_mock.Vertex)
+            else self._vertex_or_edge.graph.edges[self._vertex_or_edge.edge]
```

Comment:
Can you implement here self._vertex_or_edge.get_property(property) like in mgp.py?

Reply:
I don’t see why this needs a dedicated method in `_mgp_mock`. The behavior of this `mgp_mock` method is already consistent with `mgp` and the hidden part (`_mgp_mock`) doesn’t factor into that.